pyk/src/pyk/ktool/kprint.py

```diff
@@ -108,6 +109,8 @@ def prettyPrintKast(kast, symbol_table, debug=False):
         sys.stderr.flush()
     if kast is None or kast == {}:
         return ""
+    if type(kast) is Subst:
+        return '\n'.join(key + ' |-> ' + prettyPrintKast(value, symbol_table) for key, value in kast.items())
```

Comment:
I don't think this is a good idea, beacuse it complicates the type of `prettyPrintKast`. Please add type annotations to `prettyPrintKast`.

I'd rather it be a method of `Subst` called `pretty_print`, which takes a `KDefinition` I suppose.

Reply:
I would too, but that introduces a circular dependency between `kprint` and `kast`.  It would need to take a `KPrint` to make the recursive calls to print the values. How do you feel about moving `Subst` to its own file?

Reply:
Hmm, I've got an idea to get around this. Give me a bit.

Reply:
@ehildenb For now, I've got around the circular dependency by not typing `Subst.pretty_print`.